packages/purpose-process/src/services/purposeService.ts

```diff
@@ -733,6 +738,114 @@ export function purposeServiceBuilder(
         isRiskAnalysisValid: true,
       };
     },
+    async clonePurpose({
+      purposeId,
+      organizationId,
+      seed,
+      correlationId,
+      logger,
+    }: {
+      purposeId: PurposeId;
+      organizationId: TenantId;
+      seed: ApiPurposeCloneSeed;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> {
+      logger.info(`Cloning Purpose ${purposeId}`);
+
+      const tenant = await retrieveTenant(organizationId, readModelService);
+      assertTenantKindExists(tenant);
+
+      const purposeToClone = await retrievePurpose(purposeId, readModelService);
+
+      if (purposeIsDraft(purposeToClone.data)) {
+        throw purposeCannotBeCloned(purposeId);
+      }
+
+      const dailyCalls = getDailyCallsFromPurposeToClone(purposeToClone.data);
+
+      const newPurposeVersion: PurposeVersion = {
+        id: generateId(),
+        createdAt: new Date(),
+        state: purposeVersionState.draft,
+        dailyCalls,
+      };
+
+      const riskAnalysisFormToClone = purposeToClone.data.riskAnalysisForm;
+
+      const clonedRiskAnalysisForm: PurposeRiskAnalysisForm | undefined =
+        riskAnalysisFormToClone
+          ? {
+              id: generateId(),
+              version: riskAnalysisFormToClone.version, // TO DO double-check
+              riskAnalysisId: riskAnalysisFormToClone.riskAnalysisId,
+              singleAnswers: riskAnalysisFormToClone.singleAnswers.map(
+                (answer) => ({
+                  ...answer,
+                  id: generateId(),
+                })
+              ),
+              multiAnswers: riskAnalysisFormToClone.multiAnswers.map(
+                (answer) => ({
+                  ...answer,
+                  id: generateId(),
+                })
+              ),
+            }
+          : undefined;
+
+      const currentDate = new Date();
+      const title = purposeToClone.data.title;
+      const suffix = ` - clone - ${formatDateAndTime(currentDate)}`;
+      const prefixLengthAllowance = 60 - suffix.length - 3;
+      // 60 is the maximum length for the purpose title, according to the api spec (PurposeSeed)
+      const clonedPurposeTitle =
+        title.length + suffix.length <= 60
+          ? `${title}${suffix}`
+          : `${title.slice(0, prefixLengthAllowance)}...${suffix}`;
```

Comment:
Instead of the three dots we could use the ellipsis character

```suggestion
          : `${title.slice(0, prefixLengthAllowance)}…${suffix}`;
```

Reply:
With the three dots it's clearer that we cut by 3 additional characters. Otherwise we should cut by 1 and it might not be clear.

Reply:
Is it important to cut those 3 additional character? With the ellipsis character we could only cut 1

Reply:
But I have no strong opinion on that, so feel free to leave it like that

Reply:
I know, what I meant was: if someone doesn't know that the ellipsis doesn't count as 3 characters, they might ask "why are are we cutting just 1 character?". Just for saying.

Reply:
Updated with `const dots = "..."`